src/Oscoin/Storage/Block/SQLite.hs

```diff
@@ -78,22 +78,23 @@ lookupBlock
     => Handle c tx s
     -> BlockHash c
     -> IO (Maybe (Block c tx (Sealed c s)))
-lookupBlock Handle{hConn} h = Sql.withTransaction hConn $ do
-    result :: Maybe (BlockHeader c (Sealed c s)) <- listToMaybe <$> Sql.query hConn
-        [sql| SELECT parenthash, datahash, statehash, timestamp, difficulty, seal
-                FROM blocks
-               WHERE hash = ? |] (Only h)
+lookupBlock Handle{hConn} h = withMVar hConn $ \conn ->
```

Comment:
I suspect we would benefit from using the `Transaction` abstraction here as well. On it.